CMakeLists.txt

```diff
@@ -0,0 +1,713 @@
+option(SILO_ENABLE_FORTRAN "Enable Fortran interface to Silo" ON)
+option(SILO_ENABLE_HDF5 "Enable hdf5 support" ON)
+option(SILO_ENABLE_JSON "Enable experimental json features" OFF)
+option(SILO_ENABLE_PYTHON_MODULE "Enable python module" OFF) 
+option(SILO_ENABLE_NETCDF "Enable netcdf support" ON)
+option(SILO_ENABLE_ZFP "Enable Lindstrom array compression" ON)
+option(SILO_ENABLE_TESTS "Enable building of tests." OFF)
+option(SILO_ENABLE_TESTS_INSTALL "Enable installation of tests." OFF)
+option(SILO_BUILD_FOR_BSD_LICENSE  "Build BSD licensed version of Silo" ON)
+
+include(CMakeDependentOption)
+CMAKE_DEPENDENT_OPTION(SILO_ENABLE_FPZIP "Enable Lindstrom float 1,2,3D array compression" ON
+                       "NOT SILO_BUILD_FOR_BSD_LICENSE" OFF)
+CMAKE_DEPENDENT_OPTION(SILO_ENABLE_HZIP "Enable Lindstrom hex/quad mesh compression" ON
+                       "NOT SILO_BUILD_FOR_BSD_LICENSE" OFF)
+
+##
+#  Set up a default INSTALL prefix that is peer to the build directory
+##
+set(CMAKE_INSTALL_PREFIX ${Silo_BINARY_DIR}/../SiloInstall  CACHE PATH "install prefix" FORCE)
+
+
+set(BUILD_SHARED_LIBS ${SILO_ENABLE_SHARED})
+
+###---------------------------------------------------------------------------
+# Include any config-site <host>.cmake files if available
+#    the config-site file can override any of the default SILO_ settings
+#    as well as CMAKE_INSTALL_PREFIX
+###---------------------------------------------------------------------------
+include(SiloHelpers)
+
+# if user passed -DSILO_CONFIG_SITE to cmake, use it, otherwise
+# see if there is a config-site file for this host
+if(DEFINED SILO_CONFIG_SITE AND EXISTS ${SILO_CONFIG_SITE})
+    include(${SILO_CONFIG_SITE})
+else()
+    if(WIN32)
+        if(EXISTS ${Silo_SOURCE_DIR}/config-site/$ENV{COMPUTERNAME}.cmake)
+            include(${Silo_SOURCE_DIR}/config-site/$ENV{COMPUTERNAME}.cmake)
+        endif()
+    else()
+        if(EXISTS ${Silo_SOURCE_DIR}/config-site/$ENV{HOST}.cmake)
+            include(${Silo_SOURCE_DIR}/config-site/$ENV{HOST}.cmake)
+        elseif(EXISTS ${Silo_SOURCE_DIR}/config-site/$ENV{HOSTNAME}.cmake)
+            include(${Silo_SOURCE_DIR}/config-site/$ENV{HOSTNAME}.cmake)
+        endif()
+    endif()
+endif()
+
+###---------------------------------------------------------------------------
+# Create a custom target to copy dendencies on Windows
+###---------------------------------------------------------------------------
+
+if(WIN32 AND (SILO_ENABLE_SILEX OR SILO_ENABLE_BROWSER))
+    add_custom_target(copy_deps ALL)
+    set_target_properties(copy_deps PROPERTIES FOLDER utility)
+endif()
+
+###---------------------------------------------------------------------------
+# Find necessary packages
+###---------------------------------------------------------------------------
+
+##
+# Perl, needed to generate lite_pdb.h and lite_score.h
+##
+find_program(perl_exe NAMES perl perl.exe REQUIRED)
```

Comment:
Hmmm...I thought we committed the `lite` versions of these files to the repo meaning only developers need perl to generate them. That said, I don't think Perl is a potential problematic dependency. Does CMake have its own built-in Perl command (probably not...Perl is a big package).

Reply:
Files still need to be  generated.